controllers/sms-gateway.js

```diff
@@ -119,6 +120,7 @@ module.exports = {
       .then(JSON.parse)
       .then(function(request) {
         // Process webapp-terminating messages asynchronously
+        // TODO these promises do nothing!
```

Comment:
I get that, but because `saveToDb` is async already all this `Promise.resolve()` is doing is providing a second async layer. If you just have the loop which fires off all the `saveToDb` calls none of them will complete (or even start) before this function has returned.